File: 33-search-in-rotated-sorted-array/search-in-rotated-sorted-array.java
class Solution {
    public int search(int[] nums, int t) {
        int l=0;
        int r=nums.length-1;

        int ind=-1;

        while(l<=r){
            int mid=(l+r)/2;
             if(nums[mid]==t)return mid;

            else if(nums[mid]>=nums[l]){
            if(nums[mid]>t && nums[l]<=t ){
              r=mid-1;
            }
            else{
l=mid+1;  
            }
            }
            else{
               if(nums[mid]<t && nums[r]>=t)  {
                l=mid+1;
               }
               else{
                r=mid-1;
               }
            }
        }
        return -1;
    }
}

File: 33-search-in-rotated-sorted-array/search-in-rotated-sorted-array.cpp
class Solution {
public:
    int search(vector<int>& nums, int tar) {
        int n=nums.size();
        int idx=-1;
        int l=0,h=n-1;
        bool found=false;
        while(l<=h){
            int m=(l+h)/2;
            if(nums[m]==tar){
                idx=m;
                found=true;
            }
            if(nums[m]>=nums[l]){
              if(tar>=nums[l]&&tar<=nums[m]){
                 h=m-1;
              }
              else{
                l=m+1;
              }
            }
            else{
                if(tar>=nums[m]&&tar<=nums[h]){
                    l=m+1;
                }
                else{
                    h=m-1;
                }

            }
        }
        return (found?idx:-1);
    }
};